.github/workflows/integration.yaml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - '**/*.rst'
+      - '**/*.md'
+    branches:
+      - master
+      - '[0-9].[0-9]'
+  pull_request:
+    branches:
+      - master
+      - '[0-9].[0-9]'
+
+permissions:
+  contents: read  #  to fetch code (actions/checkout)
+
+jobs:
+
+   run-tests:
+     runs-on: ${{matrix.os}}
+     timeout-minutes: 30
+     strategy:
+       max-parallel: 15
+       matrix:
+         python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', 'pypy-3.7', 'pypy-3.8']
+         os: ['ubuntu-latest', 'windows-latest', 'macos-latest']
+       fail-fast: false
+     env:
+       ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```

Comment:
Why do you need it? I don't see any need for that as no need to run deprecated commands set-env and add-path.